aws/asg_test.go

```diff
@@ -53,7 +54,7 @@ func TestListAutoScalingGroups(t *testing.T) {
 	// clean up after this test
 	defer nukeAllAutoScalingGroups(session, []*string{&groupName})
 
-	groupNames, err := getAllAutoScalingGroups(session, region)
+	groupNames, err := getAllAutoScalingGroups(session, region, time.Now().Add(1*time.Hour))
```

Comment:
It's VERY important to check this logic actually excludes things properly!

I'd add a check where you create an instance, sleep for 10 seconds, and then call this method with the `excludeSince` param set to 1 minute ago. You can then check that the instance is NOT returned. If that works, you then set `excludeSince` to 1 hour from now and make sure it IS returned.

The same applies to the tests for all the other resources (ELBs, ASGs, etc) as well.